src/io/avro/write/schema.rs

```diff
@@ -22,20 +23,25 @@ pub fn to_record(schema: &Schema) -> Result<Record> {
     })
 }
 
-fn field_to_field(field: &Field) -> Result<AvroField> {
-    let schema = type_to_schema(field.data_type(), field.is_nullable)?;
+fn field_to_field(field: &Field, name_counter: &mut i32) -> Result<AvroField> {
+    let schema = type_to_schema(field.data_type(), field.is_nullable, name_counter)?;
     Ok(AvroField::new(&field.name, schema))
 }
 
-fn type_to_schema(data_type: &DataType, is_nullable: bool) -> Result<AvroSchema> {
+fn type_to_schema(data_type: &DataType, is_nullable: bool, name_counter: &mut i32) -> Result<AvroSchema> {
     Ok(if is_nullable {
-        AvroSchema::Union(vec![AvroSchema::Null, _type_to_schema(data_type)?])
+        AvroSchema::Union(vec![AvroSchema::Null, _type_to_schema(data_type, name_counter)?])
     } else {
-        _type_to_schema(data_type)?
+        _type_to_schema(data_type, name_counter)?
     })
 }
 
-fn _type_to_schema(data_type: &DataType) -> Result<AvroSchema> {
+fn _get_field_name(name_counter: &mut i32) -> String {
+    *name_counter += 1;
+    "r".to_owned() + &name_counter.to_string()
```

Comment:
```suggestion
    f"r{name_counter}"
```

I think this is 1 allocation vs 3 allocations on the current situation (1 for the `r`, 1 for the `name_counter.to_string()`, one for the concatenation of the two)